indexing-service/src/main/java/org/apache/druid/indexing/worker/shuffle/ShuffleResource.java

```diff
@@ -98,14 +96,19 @@ public Response getPartition(
       );
       return Response.status(Status.NOT_FOUND).entity(errorMessage).build();
     } else {
-      shuffleMetrics.ifPresent(metrics -> metrics.shuffleRequested(supervisorTaskId, partitionFile.length()));
-      return Response.ok(
-          (StreamingOutput) output -> {
-            try (final FileInputStream fileInputStream = new FileInputStream(partitionFile)) {
-              ByteStreams.copy(fileInputStream, output);
-            }
-          }
-      ).build();
+      try {
+        long size = partitionFile.get().size();
```

Comment:
This will potentially read the entire file to get the size. I believe `partitionFile.length()` is just a metadata lookup on the file to get the length.

Reply:
partitionFile is of type `Optional<ByteSource>`. The get is to get the ByteSource from the wrapping Optional class.

Reply:
I was looking at the default implementation of `ByteSource#size()` which opens the stream, reads it and counts the bytes. It looks like `FileByteSource` has overridden that implementation to return the length of the File without reading the bytes.

At this point, I'm mostly concerned about whether someone can trip over this by changing the implementation and not realizing that this operation should be fast. Anyways, that's a problem for another time.

Reply:
The `LocalIntermediaryDataManager` uses `Files.asByteSource()` which returns `FileByteSource` which do file.length() which is the same as current behavior. The extension (that implements `IntermediaryDataManager`) can extends ByteSource and provide a method of returning size efficiently. I added javadoc to indicate this